### PR TITLE
Update index.d.ts added modelOptions interface

### PR DESCRIPTION
**Pull request by Mangetsu-exe**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ declare namespace OttomanJS {
 
   type ValidatorFunction = (value: any) => void
   interface SchemaField {
-    type: 'string' | 'number' | 'integer' | 'boolean' | 'Date' | 'Mixed' | ModelReference,
+    type: 'string' | 'number' | 'integer' | 'boolean' | 'Date' | 'mixed' | ModelReference,
```
Comment (mz2) [inline index.d.ts]: I see this "mixed" change which is basically the opposite that Will noticed last night. Why from 'Mixed' to 'mixed'?

File: index.d.ts [hunk 2]
```diff
@@ -43,6 +43,20 @@ declare namespace OttomanJS {
     validate<T> (modelInstance: ModelInstance<T>, callback: ValidationCallback<T>): void
   }
 
+  interface ModelOptions {
+    index?: IndexDefinition
+    query?: QueriesDefinition
```
Comment (mz2) [inline index.d.ts]: Please be consistent with the singular form used for IndexDefinition (the terms should be QueryDefinition and Query)

**Review by mz2 — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 2 above)
Comment (Mangetsu-exe) [reply index.d.ts]: oh ok, I did that because the options from ottoman called queries and index @mz2

File: index.d.ts (hunk 1 above)
Comment (Mangetsu-exe) [reply index.d.ts]: I did not change that, weird :disappointed:  maybe by mistake somehow

**Comment by MPLtd** [comment]
Oh yeah, to be honest the other type I added has similar inconsistency to it too potentially with the terms in the source. Let’s see if the upstream maintainers are happy with these...

Matias

- - - - -
Matias Piipari, PhD
http://twitter.com/mz2

Manuscripts - A writing tool like nothing you have seen before
http://manuscriptsapp.com

**Review by mz2 — APPROVED** [review]

**Pull request merged**